setup.py

```diff
@@ -822,6 +871,7 @@ def make_relative_rpath(path):
     DL = Extension("torch._dl",
                    sources=["torch/csrc/dl.c"],
                    language='c',
+                   extra_link_args=['-shared'] if WITH_ROCM else []
```

Comment:
Why does `ROCM` affect this one? It's completely unrelated to any GPU stuff

Reply:
On the ROCm stack, the HCC compiler generates both the host & device code. I briefly spoke with a few developers at AMD about allowing other compilers such as GCC to be used on host - though it's not on their road map. It'll require modifications to [HIPCC](https://github.com/ROCm-Developer-Tools/HIP/blob/master/bin/hipcc). I'll give it a dive soon.

(HIPCC is what we set as the compiler in CMakeLists.txt)